Sources/ProjectSpec/Breakpoint.swift

```diff
@@ -0,0 +1,254 @@
+        case `throw` = "0"
+        case `catch` = "1"
+
+        init?(_ text: String) {
+            let text = text.lowercased()
+            switch text {
+            case "throw":
+                self = .throw
+            case "catch":
+                self = .catch
+            default:
+                return nil
+            }
+        }
+    }
+
+    public struct Action: Equatable {
+
+        public enum ConveyanceType: String, Equatable {
+            case console = "0"
+            case speak = "1"
+
+            init?(_ text: String) {
+                let text = text.lowercased()
+                switch text {
+                case "console":
+                    self = .console
+                case "speak":
+                    self = .speak
+                default:
+                    return nil
+                }
+            }
+        }
+
+        public enum SoundName: String, Equatable {
+            case basso = "Basso"
+            case blow = "Blow"
+            case bottle = "Bottle"
+            case frog = "Frog"
+            case funk = "Funk"
+            case glass = "Glass"
+            case hero = "Hero"
+            case morse = "Morse"
+            case ping = "Ping"
+            case pop = "Pop"
+            case purr = "Purr"
+            case sosumi = "Sosumi"
+            case submarine = "Submarine"
+            case tink = "Tink"
+        }
+
+        public var type: BreakpointActionType
+        public var consoleCommand: String?
+        public var message: String?
+        public var conveyanceType: ConveyanceType?
+        public var command: String?
+        public var arguments: String?
+        public var waitUntilDone: Bool?
+        public var script: String?
+        public var soundName: SoundName?
+
+        public init(type: BreakpointActionType,
+                    consoleCommand: String? = nil,
+                    message: String? = nil,
+                    conveyanceType: ConveyanceType? = nil,
+                    command: String? = nil,
+                    arguments: String? = nil,
+                    waitUntilDone: Bool? = nil,
+                    script: String? = nil,
+                    soundName: SoundName? = nil) {
+            self.type = type
+            self.consoleCommand = consoleCommand
+            self.message = message
+            self.conveyanceType = conveyanceType
+            self.command = command
+            self.arguments = arguments
+            self.waitUntilDone = waitUntilDone
+            self.script = script
+            self.soundName = soundName
+        }
+    }
+
+    public var type: BreakpointType
+    public var enabled: Bool
+    public var ignoreCount: Int
+    public var continueAfterRunningActions: Bool
+    public var filePath: String?
+    public var timestamp: String?
+    public var line: Int?
+    public var breakpointStackSelectionBehavior: String?
+    public var symbol: String?
```

Comment:
To make the API nicer, we could possibly also change the breakpoint type to a custom enum with associated types. This will make creating breakpoints in Swift much friendlier. Not required here in this PR but would be nice. We'd have to be careful to not make incorrect assumptions about which properties belong with which types. What do you think? 

eg:
```swift
type: .file(path: "File.swift", line: 58)
```

Reply:
I thought about it. 
However, I didn't know if I should do it.

I would be glad to do it in this PR.
But it might take some time since I am a little busy with my work.
Is that alright?